### ISSUE TITLE
ipad 12.9 ios 14.0.1 Webgl is not supported

### ISSUE THREAD

**Issue by ingo916** [jneilliii/OctoPrint-BedLevelVisualizer#311]
ipad 12.9 ios 14.0.1 error Webgl is not supported by your browser - visit http://get.webgl.org for more info
Works fine with my iphone ios 14.0.1

visiting the link says my browser is supported and a rotating cube is displayed.

**Debug Log**
will include this log soon.

**Screenshots**
![Image](https://user-images.githubusercontent.com/41309095/95894073-74486600-0d3d-11eb-84bd-637a961a4dd6.png)
![Image-1](https://user-images.githubusercontent.com/41309095/95894081-76122980-0d3d-11eb-94cd-763f24b19e67.png)

**Firmware and Version**
OctoPrint 1.4.2 Python 3.7.3 OctoPi 0.17.0

**Comment by jneilliii** [comment]
This is probably just a detection issue with the version of plotly js library that is installed with the plugin. I'll have to see if there is a newer version upstream that recognizes your device properly.

**Comment by jneilliii** [comment]
do you have the temps graph plugin installed or any other plugin that includes graphing features? I just looked and the version of plotly that is included with the plugin should already be patched for iPad 12.9 so not sure if it's a ios version issue or safari version issue possibly. 

https://github.com/plotly/plotly.js/issues/4502

**Comment by ingo916** [comment]
These are the plugins I have installed
BLTouch
Backup Scheduler
Bed Visualizer
GcodeEditor
Marlin EEPROM Editor
NavBarTemp
octolapse
Preheat Button
Printer Notifications
Pushover
SpoolManager
TP-Link Smartplug
Themeify
Virtual Printer



Ingo Azarvand
209-207-2243
Licensed FFL Dealer

**Comment by jneilliii** [comment]
Lol, images don't come over in email replies.

**Comment by ingo916** [comment]
> 
> 
> Lol, images don't come over in email replies.

BLTouch
Backup Scheduler
Bed Visualizer
GcodeEditor
Marlin EEPROM Editor
NavBarTemp
octolapse
Preheat Button
Printer Notifications
Pushover
SpoolManager
TP-Link Smartplug
Themeify
Virtual Printer

**Comment by jneilliii** [comment]
Ah, I might need to check tplink. If you disable that plugin temporarily will this plugin work for you?

**Comment by ingo916** [comment]
disabling the TPLink plugin worked! Please let me know if you have a solution that doesn't require disabling the TPLink plugin.

**Comment by jneilliii** [comment]
I just need to add the latest plotly library to that plugin. I can do that tonight when I get home and provide a link you can use to install a pre-release version to use.

**Comment by jneilliii** [comment]
I've added the same plotly version to TPLink plugin. Install it by copying/pasting the URL below into plugin manager > Get More > ...from URL and dclick Install.

```
https://github.com/jneilliii/OctoPrint-TPLinkSmartplug/archive/0.9.25.zip
```

**Comment by ingo916** [comment]
Thanks the updated TPLinkSmartplug update you provided resolved my issue. Its all working great now!!!

**Comment by ingo916** [comment]
FYI after installing the 0.9.25.zip i can no longer manage my plug using the plugin. I deleted the plug hoping that i just needed to recreate it and I cannot. tried adding the plug and nothing happens. I uninstalled the plugin and removed all data, reinstalled using the link you provided and still cannot add a plug. I included a video attached.


[2020-10-14_12-58-34.zip](https://github.com/jneilliii/OctoPrint-BedLevelVisualizer/files/5380570/2020-10-14_12-58-34.zip)
